lib/services/consumers/application.dao.js

```diff
@@ -51,59 +70,73 @@ dao.findAll = function (query) {
   });
 };
 
+dao.find = function (appName) {
+  return db.smembersAsync(appNameSetKey(appName))
+    .then(function (Ids) {
+      if (Ids && Ids.length !== 0) {
+        if (Ids.length === 1) {
+          return Ids[0];
+        }
+        throw new Error(`Multiple applications with ${appName} have been found: ${Ids.join(',')}.
+                         Please search for it using its ID instead`);
+      } else {
+        return false;
+      }
+    });
+};
+
 dao.get = function (id) {
-  return db.hgetallAsync(config.systemConfig.db.redis.namespace.concat('-', appNamespace).concat(':', id))
-  .then(function (app) {
-    if (!app || !Object.keys(app).length) {
-      return false;
-    } else return app;
-  });
+  return db.hgetallAsync(appHashKey(id))
+    .then(function (app) {
+      if (!app || !Object.keys(app).length) {
+        return false;
+      } else return app;
+    });
 };
 
 dao.getAll = function (userId) {
   return this.getAllAppIdsByUser(userId)
-  .then(appIds => {
-    return Promise.all(appIds.map(this.get))
-    .then(apps => apps.filter(app => app !== false));
-  });
+    .then(appIds => {
+      return Promise.all(appIds.map(this.get))
+        .then(apps => apps.filter(app => app !== false));
+    });
 };
 
 dao.getAllAppIdsByUser = function (userId) {
-  return db.smembersAsync(config.systemConfig.db.redis.namespace.concat('-', userAppsNamespace).concat(':', userId));
+  return db.smembersAsync(userAppsHashKey(userId));
 };
 
 dao.activate = function (id) {
-  return db.hmsetAsync(config.systemConfig.db.redis.namespace.concat('-', appNamespace).concat(':', id), ['isActive', 'true', 'updatedAt', String(new Date())]);
+  return db.hmsetAsync(appHashKey(id), ['isActive', 'true', 'updatedAt', String(new Date())]);
 };
 
 dao.deactivate = function (id) {
-  return db.hmsetAsync(config.systemConfig.db.redis.namespace.concat('-', appNamespace).concat(':', id), ['isActive', 'false', 'updatedAt', String(new Date())]);
+  return db.hmsetAsync(appHashKey(id), ['isActive', 'false', 'updatedAt', String(new Date())]);
 };
 
 dao.deactivateAll = function (userId) {
   return this.getAllAppIdsByUser(userId)
-  .then(appIds => {
-    const deactivateAppPromises = appIds.map(appId => this.deactivate(appId));
-    return Promise.all(deactivateAppPromises);
-  });
+    .then(appIds => {
+      const deactivateAppPromises = appIds.map(appId => this.deactivate(appId));
+      return Promise.all(deactivateAppPromises);
+    });
 };
 
-dao.remove = function (id, userId) {
+dao.remove = function ({ name, id }, userId) {
```

Comment:
I saw in the app service you are passing entire `app` here
so `{name, id, userId}` should work 

as result in dao, `remove` will accept single argument - app

Reply:
That's true. I'll see if I can rearrange this part.